src/dialog-user/services/dialogData.ts

```diff
@@ -45,21 +56,26 @@ export default class DialogDataService {
    * @param data {any} This is a object that is all the information for a particular dialog field
    *
    **/
-  private updateFieldSortOrder(data) {
-    const SORT_DESCRIPTION = 1;
-    const SORT_VALUE = 0;
-    const FIRST_OPTION = 0;
-    const VALUE = 0;
-    const sortBy = (data.options.sort_by === 'value' ? SORT_VALUE : SORT_DESCRIPTION);
-    let tempValues = [...data.values];
+  private updateFieldSortOrder({
+    options,
+    values,
+  }) {
+    let tempValues = [...values];
+
+    // The following deals with an empty default option if it exists
+    const firstValue = values[0][0];
     let defaultDropdownField = [];
-    // The following if deals with a empty default option if it exists
-    if (data.data_type === 'integer' && _.isNaN(tempValues[FIRST_OPTION][VALUE]) ||
-      _.isNull(tempValues[FIRST_OPTION][VALUE])) {
+    if (! firstValue) {
```

Comment:
TODO: test fails here because this also works for actual zeroes now

but, we don't have NaN after the conversion anymore, only zeroes or null
should we convert non-numeric strings to null instead of 0 when converting to integer?

Reply:
Hmm, what do you think makes more sense? I think in either case the sort order is just going to be messed up anyway because if you have multiple things that convert into 0 or null, it will probably just keep the order they're already in. I think maybe we convert into 0 simply so that it's "guaranteed" that they get an integer, since that is what they asked for by setting the data_type to integer (even if it wasn't meant to be that).

Reply:
Well, this whole bit of code is to support options like `{nil => "Nothing selected", 123 => "Default"}`,
and it only looks at the first entry.

So.. the only thing this affects is pretty much that - if you first value is 0, it will end up always getting sorted first, whereas previously only `""`, `null` and `"non-numeric string"` would get treated like that.

Reply:
Huh.. and I still have to fix `""` - that should be treated as a `null` also, right?

Reply:
@eclarizio WDYT? Should we always treat `""` as `null` or only for integers?

Reply:
@himdel Alright, that makes sense. As far as `""` goes, for an integer I believe on the backend we would be treating it as a "0", it's only if it's like you had above (`{nil => "nothing selected"}`) where it would stay as `nil`. https://github.com/ManageIQ/manageiq/blob/master/app/models/dialog_field_sorted_item.rb#L113

Reply:
For the failing spec I guess we have 2 options:

* start treating 0 as a potential default value with integers, fix the spec (it will pass if I replace 0 with 1)
* convert non-numeric strings to null instead of 0

`0` -

Reply:
I think it's potentially feasible for someone to actually create a hash that has `{0 => "Zero", 1 => "One", 2 => "Two"} #...etc`, so maybe treating 0 as a potential default value with integers and fixing the spec is the way to go. It will also be easier to explain to a customer that "hey, you set the `data_type` to integer so that's why your "foo"s are getting converted to "0"s.

Reply:
You're right, that seems to be the safest option :)